### ISSUE TITLE
Build intermittently fails after 600 seconds with "Can`t build - status is not pending"

### ISSUE THREAD

**Issue by jwmwalrus** [php-censor/php-censor#205]
Hi.

When trying to run some functional tests, using headless chromium at port 9222 and Symfony's web server (with specific router) at port 8000, the test intermittently fails after 600 seconds with "Can`t build - status is not pending".

Moreover, the build log stops echoing output right after the headless chromium instance is started through supervisor.

Since the build fails intermittently, it also succeeds intermittently, for the same commit.

Am I missing something?

## project configuration

The configuration used for the builds follows.  Comments were added for each invocation of the project's setup script 
```yaml
build_settings:
    ignore:
        - "app"
        - "bin"
        - "node_modules"
        - "setup-data"
        - "var"
        - "vendor"
        - "web"

setup:
    composer:
        action: "show"
    shell:
        - "git submodule foreach git checkout %BRANCH%"
        - "cp /var/www/.ci-data/myproject/parameters.yml %BUILD_PATH%/app/config/parameters.yml"
        - "php %BUILD_PATH%/setup ci"    #composer install && npm ci
        - "php %BUILD_PATH%/setup testdb"   #setup test database (MySQL)
        - "php %BUILD_PATH%/setup runprod"   #run webpack build
        - "php %BUILD_PATH%/setup testports"   #make sure ports 9222 and 8000 are free
        - "supervisorctl start headless"
        - "php %BUILD_PATH%/bin/console server:start --router=web/test_router.php localhost:8000"

test:
    php_unit:
        config:
            - "phpunit.xml.dist"
        coverage: "tests/reports/coverage"
        args: "--testsuite MYTESTSUITE"

complete:
    shell:
        - "php %BUILD_PATH%/bin/console server:stop"
        - "supervisorctl stop headless"
    copy_build:
        directory: "/var/www/.ci-data/myproject/last-build"
        respect_ignore: false
        wipe: true
```

## Environment

* PHP Censor version: v0.22.0
* Operating System: Debian GNU/Linux 9.5 (stretch)
* PHP version: 7.2.7-1+0\~20180622080745.23+stretch~1.gbpfd8e2e
* MySQL/PostgreSQL version: 15.1 Distrib 10.1.26-MariaDB

**Comment by corpsee** [comment]
Hi @jwmwalrus. Thank you for detailed report!

The problem same like #190. This timeout removed in `master` branch. Can you try to use `master` branch instead of version `v0.22.0`?

**Comment by jwmwalrus** [comment]
I should have used the git description instead of the tag.

```bash
root@git:/var/www/php-censor# git branch
* master
root@git:/var/www/php-censor# git describe
0.22.0-62-gbfe6ff19
```

So it seems my copy of master is only missing the two commits made today ---I only discovered php-censor a few days ago, so when I cloned the repo for the first time, the worker branch mentioned in #190 had already been merged into master.

**Comment by jwmwalrus** [comment]
I changed the value of `php-censor.queue.lifetime` to `6000` (since the test stage can take upto 90 minutes for one of the projects), and will report back if the issue of intermittent failure persists.

There's still the issue of the build log not showing anything after the ` supervisorctl start headless` command is executed ---so after the first 300 seconds of the build, I will have to wait the whole `php-censor.queue.lifetime` number of seconds to know whether the build succeeded or not.

Does PHP Censor require the port 8000 for something?

**Comment by corpsee** [comment]
As I know port 8000 isn't used anywhere.

**Comment by jwmwalrus** [comment]
The build started after I set `php-censor.queue.lifetime` to `6000` failed (after 6000 seconds) with `Can't build - status is not pending`.

After setting the `php-censor.queue.lifetime` value back to `600`, changing the port to 8002 and , I've got the same `Can't build - status is not pending`.

I'm wondering if Symfony's `bin/console server:start` mode of execution (i.e., in background) is causing an issue with the Shell plugin.
